doc/languages-frameworks/javascript.section.md

```diff
@@ -375,6 +373,54 @@ Assuming the following directory structure, we can define `sourceRoot` and `pnpm
   pnpmRoot = "frontend";
 ```
 
+#### PNPM Workspaces {#javascript-pnpm-workspaces}
+
+If you need to use a PNPM workspace for your project, then set `pnpmWorkspace = "<workspace project name>"` in your `pnpm.fetchDeps` call,
+which will make PNPM only install dependencies for that workspace package.
+
+For example:
+
+```nix
+...
+pnpmWorkspace = "@astrojs/language-server";
+pnpmDeps = pnpm.fetchDeps {
+  inherit (finalAttrs) pnpmWorkspace;
+  ...
+}
+```
+
+The above would make the fetchDeps call only install dependencies for the `@astrojs/language-server` workspace package.
+Note that you do not to set `sourceRoot` to make this work.
+
+Usually in such cases, you'd want to use `pnpm --filter=$pnpmWorkspace build` to build your project, as `npmHooks.npmBuildHook` will probably won't work. A `buildPhase` based on the following example will probably fit most workspace projects:
```

Comment:
Why can't we detect pnpmWorkspace and set that automatically in the hook?

Reply:
> Why can't we detect pnpmWorkspace and set that automatically in the hook?

See my comment regarding the `pnpm.buildHook` idea below.